pkg/frontend/tae_init.go

```diff
@@ -603,8 +603,8 @@ func FillInitialDataForMoUser() *batch.Batch {
 	return PrepareInitialDataForSchema(schema, data)
 }
 
-// InitDB setups the initial catalog tables in tae
-func InitDB(ctx context.Context, tae engine.Engine) error {
+// InitTAE setups the initial catalog tables in tae
+func InitTAE(ctx context.Context, tae engine.Engine) error {
```

Comment:
I think the 'InitTAE' is inappropriate. What you are doing is adds an abstract above the TAE. It denotes that frontend will only see the engine interface and the distributed transaction interface and will not see the tae anymore.

Reply:
Maybe just the storage or something else that can indicate the meaning.

Reply:
I think this function is TAE-specific since a type assertion guards it.
The memory engine doesn't need this to initialize its catalog.

> I think the 'InitTAE' is inappropriate. What you are doing is adds an abstract above the TAE. It denotes that frontend will only see the engine interface and the distributed transaction interface and will not see the tae anymore.

I think this function is TAE-specific since a type assertion guards it.
The memory engine doesn't need this to initialize its catalog.

Reply:
No, it constructs and checks some catalog tables on engines. It is necessary.